great_tables/_export.py

```diff
@@ -0,0 +1,246 @@
+    required packages:
+
+    ```bash
+    pip install great_tables[extra]
+    ```
+
+    One of the arguments for saving is `scale=`, which defaults to a scale value of `1.0`. This
+    default provides adequate image quality for most use cases given that text and lines are
+    rendered clearly. However, if you need a higher resolution, you can increase the `scale` level.
+    Keep in mind that file sizes will increase quite a bit with higher scale levels. At the default
+    scale level you should expect that `GT(exibble).save("exibble.png")` will produce a file of
+    about 150 KB. At a scale level of `2.0`, the file size will be about 320 KB; on the flipside, a
+    scale level of `0.5` yields a ~70 KB image.
+
+    The `expand=` argument adds whitespace pixels around the cropped table image. This has a default
+    value of `5`. This can be increased to capture more of the surrounding area, or decreased to
+    capture less (to a natural limit of `0`). The table is always captured on a white background, so
+    the `expand=` value can be useful to add some padding around the table in the image file.
+    """
+
+    # Import the required packages
+    _try_import(name="selenium", pip_install_line="pip install selenium")
+    _try_import(name="PIL", pip_install_line="pip install pillow")
+
+    from selenium import webdriver
+    from selenium.webdriver.common.by import By
+    from PIL import Image
+    from io import BytesIO
+    from pathlib import Path
+
+    Image.MAX_IMAGE_PIXELS = None
+
+    # Get the file extension from the filename
+    file_extension = filename.split(".")[-1]
+
+    # If there is no file extension, add the .png extension
+    if len(file_extension) == len(filename):
+        filename += ".png"
+        file_extension = "png"
+
+    # Accepted file extensions
+    accepted_extensions = ["png", "gif", "bmp", "tif", "tiff", "pdf"]
+
+    # If the file extension provided is not in the accepted list, raise an error
+    if file_extension not in accepted_extensions:
+        raise ValueError(
+            """The file extension provided is not supported.
+                         - please use one of the following file extensions: \
+                            "png", "gif", "bmp", "tif", "tiff", or "pdf"."""
+        )
+
+    # Map file extensions to the corresponding PIL format
+    pil_formats = {
+        "png": "png",
+        "gif": "gif",
+        "bmp": "bmp",
+        "tif": "tiff",
+        "tiff": "tiff",
+        "pdf": "pdf",
+    }
+
+    # Get the PIL format from the file extension
+    pil_format = pil_formats[file_extension]
+
+    # Get the HTML content from the displayed output
+    html_content = as_raw_html(self=self)
+
+    # Create a temp directory to store the HTML file
+    temp_dir = tempfile.mkdtemp()
+
+    # Create a temp file to store the HTML file; use the .html file extension
+    temp_file = tempfile.mkstemp(dir=temp_dir, suffix=".html")
```

Comment:
We have to manually delete this file when we're done with it (unfortunately this detail is buried at the bottom of `tempfile.mkstemp()` 😭 )

You can use a context handler to do it automatically. This is important, because if the code errors, the context handler will still make sure to delete the file...

```
with tempfile.TemporaryFile() as fp:
    fp.write(b'Hello world!')
    fp.seek(0)
    fp.read()
```

Note that the file gets deleted after the `with` block, so everything using the file needs to go in there.